internal/controller/helmrelease_controller.go

```diff
@@ -231,7 +231,8 @@ func (r *HelmReleaseReconciler) reconcile(ctx context.Context, hr v2.HelmRelease
 	}
 
 	// Check chart readiness
-	if hc.Generation != hc.Status.ObservedGeneration || !apimeta.IsStatusConditionTrue(hc.Status.Conditions, meta.ReadyCondition) {
+	if hc.Generation != hc.Status.ObservedGeneration ||
+		!apimeta.IsStatusConditionTrue(hc.Status.Conditions, meta.ReadyCondition) || hc.GetArtifact() == nil {
```

Comment:
Given it should really be an edge case, doing this in one place (before attempting to load) should be sufficient IMHO.